fmcg_store/data/ir_ui_view.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="pos_order_form_customization_fmcg_store" model="ir.ui.view">
+        <field name="name">pos.order.form.customization.fmcg</field>
+        <field name="model">pos.order</field>
+        <field name="inherit_id" ref="point_of_sale.view_pos_pos_form"/>
+        <field name="mode">extension</field>
+        <field name="priority">160</field>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <data>
+                <xpath expr="//field[@name='full_product_name']" position="after">
+                    <field name="x_best_before_days" optional="show"/>
+                </xpath>
+            </data>
+        </field>
+    </record>
+    <record id="product_product_form_customization_fmcg_store" model="ir.ui.view">
+        <field name="name">product.product.form.customization.fmcg</field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view"/>
+        <field name="mode">extension</field>
+        <field name="priority">160</field>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <data>
+                <xpath expr="//field[@name='active']" position="after">
+                    <field name="x_best_before_day"/>
+                </xpath>
```

Comment:
Shouldn't it be invisible as soon as its type is different from a consumable/storable? And I'd move it below the type
```suggestion
                <xpath expr="//field[@name='detailed_type']" position="after">
                    <field name="x_best_before_day" invisible="type not in ('consu', 'product')'/>
                </xpath>
```